src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java

```diff
@@ -252,21 +281,55 @@ public void onBlock(org.ethereum.core.Block block, List<TransactionReceipt> rece
             pegoutTxs.forEach(BtcReleaseClient.this::onBtcRelease);
         }
 
+        /**
+         * Determines if the svp spend transaction hash is ready to be signed based on its block confirmations.
+         *
+         * <p>
+         * This method retrieves the block associated with the given transaction hash and calculates
+         * the difference in block numbers between the current block and the block containing the transaction.
+         * If the difference meets or exceeds the required confirmation threshold defined in the bridge constants,
+         * the transaction is considered ready for signing.
+         * </p>
+         *
+         * @param currentBlockNumber the current block number in the blockchain
+         * @param svpTxHash the Keccak256 hash of the svp spend transaction waiting to be signed
+         * @return {@code true} if the transaction has the required number of confirmations and is ready to be signed;
+         *         {@code false} otherwise
+         */
+        private boolean isSVPSpendTxReadyToSign(long currentBlockNumber, Keccak256 svpTxHash) {
```

Comment:
not sure about this method needing to receive two more arguments in the constructor

Reply:
Only way of accomplishing what it needs to do.

Reply:
Can't we use the `releaseCreationInformationGetter` to accomplish this? It's already part of this class and it's what we use to get the info for pegouts

Reply:
Nice, thanks :) 

Updated to use `releaseCreationInformationGetter`, let me know if this looks good now.